### ISSUE TITLE
utils.community_detection clustering funtion throws error with default arguments.  

### ISSUE THREAD

**Issue by dingusagar** [UKPLab/sentence-transformers#1250]
```py
sentences = [
    'i would like to buy this',
    'want to buy this',
    'buy this',
    'wanna buy this product'
    'i want the red color',
    'i want the blue color',
]
embeddings = model.encode(sentences, batch_size=64, show_progress_bar=True, convert_to_tensor=True)
clusters = util.community_detection(embeddings, min_community_size=3, threshold=0.5)
```

the above piece of code doesn't work due to this exception.
```
RuntimeError                              Traceback (most recent call last)

<ipython-input-25-083ba576c2a6> in <module>
      6 #min_cluster_size: Only consider cluster that have at least 25 elements
      7 #threshold: Consider sentence pairs with a cosine-similarity larger than threshold as similar
----> 8 clusters = util.community_detection(embeddings, min_community_size=3, threshold=0.5)
      9 
     10 print("Clustering done after {:.2f} sec".format(time.time() - start_time))

/media/dingusagar/Data/python-envs/pytorch-env/lib/python3.7/site-packages/sentence_transformers/util.py in community_detection(embeddings, threshold, min_community_size, init_max_size)
    354 
    355             # Only check top k most similar entries
--> 356             top_val_large, top_idx_large = cos_scores[i].topk(k=init_max_size, largest=True)
    357             top_idx_large = top_idx_large.tolist()
    358             top_val_large = top_val_large.tolist()

RuntimeError: selected index k out of range
```

This is because for smaller number of sentences the default value `init_max_size=1000` doesn't work. 1000 will give index out of bound error while doing 
`cos_scores[i].topk(k=init_max_size, largest=True)`

currently my hack to fix this was 
```py
clusters = util.community_detection(embeddings, min_community_size=3, threshold=0.5, init_max_size=len(embeddings))
```
If this fix makes sense, I can create a PR that initializes init_max_size like this by default inside the community_detection function.

**Comment by nreimers** [comment]
Thanks for reporting it.

Yes, a PR would be need. As you cannot call a function as default value, but you can add a line like this:
``` 
init_max_size = min(init_max_size, len(embeddings))
```

To the function itself.

**Comment by dingusagar** [comment]
@nreimers,  added the fix in this PR #1253 and tested. please review and suggest if any further changes are required. Thanks
